src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java

```diff
@@ -188,18 +201,19 @@ private boolean cannotHoldMoreItems()
      */
     private boolean gatherFromBuilding(@NotNull final AbstractBuilding building)
     {
-        if (currentSlot >= building.getTileEntity().getSizeInventory())
+        final ItemStack stack = currentSlot >= building.getTileEntity().getSizeInventory() ? null : building.getTileEntity().getStackInSlot(currentSlot);
+
+        if (ItemStackUtils.isEmpty(stack))
         {
             return true;
         }
 
-        final ItemStack stack = building.getTileEntity().getStackInSlot(currentSlot);
-        if (ItemStackUtils.isEmpty(stack) || workerRequiresItem(building, stack, alreadyKept)
+        if (workerRequiresItem(building, stack, alreadyKept)
                 || (building instanceof BuildingHome && stack.getItem() instanceof ItemFood))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "stack" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2259)